c7n/resources/asg.py

```diff
@@ -226,14 +226,25 @@ class LaunchConfigFilter(ValueFilter):
     schema = type_schema(
         'launch-config', rinherit=ValueFilter.schema)
     schema_alias = False
-    permissions = ("autoscaling:DescribeLaunchConfigurations",)
+    permissions = ("autoscaling:DescribeLaunchConfigurations", "ec2:DescribeLaunchTemplates")
 
     def process(self, asgs, event=None):
         self.launch_info = LaunchInfo(self.manager).initialize(asgs)
         return super(LaunchConfigFilter, self).process(asgs, event)
 
     def __call__(self, asg):
-        return self.match(self.launch_info.get(asg))
+        launch_details = self.launch_info.get(asg)
+        if not launch_details:
+            return False
+
+        # In launch-config, AssociatePublicIpAddress its in the original dict
+        # For launch-template we need to look a level deeper
+        if 'NetworkInterfaces' in launch_details:
```

Comment:
this type of hardcoding to a specific key feels a bit odd, isn't the right thing to address at the policy level, by detecting launch config vs template in an or block.

```
- or:
   - and:
      - launch-config is template
      - network key
   - and:
      - launch-config is config
      - other network key
      ```